jsettlers.logic/src/main/java/jsettlers/logic/map/grid/objects/MapObjectsManager.java

```diff
@@ -400,18 +401,14 @@ public boolean pushMaterial(short x, short y, EMaterialType materialType) {
 		}
 	}
 
-	public ShortPoint2D pushMaterialForced(short x, short y, EMaterialType materialType) {
-		IMapArea mapArea = new HexGridArea(x, y, 0, 200);
-
-		for (ShortPoint2D position : mapArea) {
-			short currX = position.x;
-			short currY = position.y;
+	public ShortPoint2D pushMaterialForced(int x, int y, EMaterialType materialType) {
+		return HexGridArea.iterate(x, y, 0, 200, (currX, currY, radius) -> {
 			if (grid.isInBounds(currX, currY) && canForcePushMaterial(currX, currY, materialType)) {
 				pushMaterial(currX, currY, materialType);
-				return position;
+				return Optional.of(new ShortPoint2D(currX, currY));
 			}
-		}
-		return null;
+			return Optional.empty();
+		}).orElseGet(() -> null);
```

Comment:
@michaelzangl: Thanks for the hint. I must have overlooked that.
@codingberlin: I know. Actually I don't want to keep it this way. But at the moment my focus is on optimizing speed and getting away from these iterators creating tons of objects. So it's your second point that I see here at the moment. I just don't want to change all APIs with this change.